Dockerfile

```diff
@@ -1,8 +1,47 @@
-FROM node:14
+FROM node:16-alpine3.15
+
+# Expose gatsby + vscode debug ports
+EXPOSE 8000 9929
+
+# Set working directory
 WORKDIR /app
-COPY ./ /app/
-RUN yarn
-VOLUME "/src" /app/src
-WORKDIR "/app"
-EXPOSE 8000
-CMD ["yarn", "start"]
+
+# Prepare container to install gatsby packages with yarn
+# Ref: https://hub.docker.com/r/futureys/gatsby/dockerfile
+RUN apk add --no-cache \
+# git: gatsby new command depends on git
+    git \
+# util-linux: gatsby develop calls lscpu
+# openssl, sudo: gatsby develop --https uses
+    util-linux openssl sudo \
+# node-sess version 4 uses g++ and still requires python2
+# see: https://github.com/sass/node-sass/issues/2877
+    python3 g++ \
+# gatsby-plugin-sharp depends on imagemin-mozjpeg,
+# imagemin-mozjpeg depends on mozjpeg,
+# mozjpeg requires compiling from source with autoreconf, automake, libtool, gcc, make, musl-dev, file, pkgconfig, nasm
+# see: https://pkgs.alpinelinux.org/contents?page=1&file=autoreconf
+# see: https://github.com/buffer/pylibemu/issues/24#issuecomment-492759639
+# see: https://github.com/maxmind/libmaxminddb/issues/9#issuecomment-30836272
+# see: https://stackoverflow.com/questions/28631817/no-acceptable-c-compiler-found-in-path/57419374#57419374
+# see: https://pkgs.alpinelinux.org/contents?branch=v3.3&name=file&arch=x86&repo=main
+# see: https://stackoverflow.com/questions/17089858/pkg-config-pkg-prog-pkg-config-command-not-found/17106579#17106579
+# see: https://github.com/alicevision/AliceVision/issues/593#issuecomment-457194176
+    autoconf automake libtool gcc make musl-dev file pkgconfig nasm \
+# sharp depends on vips
+# see: https://github.com/lovell/sharp/issues/773
+    vips \
+ && rm -fR /var/cache/apk/*
```

Comment:
thx for chasing down these guys. When I was putting together the file I decided against alpine for ease of use, but am happy to use it now :)

Reply:
Yeah I'm not beholden to alpine long term or anything so we could always file a future enhancement/issue if need be